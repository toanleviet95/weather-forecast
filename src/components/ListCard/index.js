import React from 'react';
import PropTypes from 'prop-types';
import Card from '@/components/Card';
import { DEFAULT_WEATHERS } from '@/shared/constants';

const ListCard = ({ data = { consolidated_weather: DEFAULT_WEATHERS, location: '' }, loading }) => {
  const { consolidated_weather: weathers, title: location } = data;

  // Only max of length with 5
  const requiredWeathers = weathers.slice(0, 5);

  return (
    <div className='forecast-table row'>
      <div className='forecast-container'>
        {requiredWeathers.map((weather, idx) =>
        (
          <div key={`weather_${weather.id}`} className={`forecast ${idx === 0 ? 'today' : 'col-md-3 col-xs-12'}`}>
            <Card
              isToday={idx === 0}
              loading={loading}
              date={weather.applicable_date}
              location={location}
              low={weather.min_temp}
              high={weather.max_temp}
              temp={weather.the_temp}
              state={weather.weather_state_name}
            />
          </div>
        ))}
      </div>
    </div>
  );
};

ListCard.propTypes = {
  data: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default ListCard;
