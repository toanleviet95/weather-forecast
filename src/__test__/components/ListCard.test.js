import React from 'react';
import renderer from 'react-test-renderer';
import ListCard from '@/components/ListCard';

describe('ListCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ListCard />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
