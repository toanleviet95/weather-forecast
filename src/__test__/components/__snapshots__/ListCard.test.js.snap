// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard renders correctly 1`] = `
<div
  className="forecast-table row"
>
  <div
    className="forecast-container"
  >
    <div
      className="forecast today"
    >
      <div
        className="forecast-header"
      >
        <div
          className="day"
        >
          --
           
          (Today)
        </div>
        <div
          className="date"
        >
          --
        </div>
      </div>
      <div
        className="forecast-content"
      >
        <div
          className="location"
        >
          <h3
            id="city"
          >
            --
          </h3>
        </div>
        <div
          className="degree"
        >
          <div
            className="num"
            id="tempnow"
          >
            --
            <sup>
              o
            </sup>
            C
          </div>
          <div
            className="forecast-icon"
          >
            <img
              src=""
              width={90}
            />
          </div>
        </div>
        <span
          id="windnow"
        >
          Low:
           
          --
          <sup>
            o
          </sup>
          C
        </span>
        <span
          id="directionnow"
        >
          High:
           
          --
          <sup>
            o
          </sup>
          C
        </span>
      </div>
    </div>
    <div
      className="forecast col-md-3 col-xs-12"
    >
      <div
        className="forecast-header"
      >
        <div
          className="day"
        >
          --
        </div>
        <div
          className="date"
        >
          --
        </div>
      </div>
      <div
        className="forecast-content"
      >
        <div
          className="forecast-icon"
        >
          <img
            src=""
            width={48}
          />
        </div>
        <div
          className="degree"
        >
          --
          <sup>
            o
          </sup>
          C
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12 col-md-offset-0 col-xs-4 col-xs-offset-2"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              Low:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
          <div
            className="col-md-12 col-xs-4"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              High:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="forecast col-md-3 col-xs-12"
    >
      <div
        className="forecast-header"
      >
        <div
          className="day"
        >
          --
        </div>
        <div
          className="date"
        >
          --
        </div>
      </div>
      <div
        className="forecast-content"
      >
        <div
          className="forecast-icon"
        >
          <img
            src=""
            width={48}
          />
        </div>
        <div
          className="degree"
        >
          --
          <sup>
            o
          </sup>
          C
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12 col-md-offset-0 col-xs-4 col-xs-offset-2"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              Low:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
          <div
            className="col-md-12 col-xs-4"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              High:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="forecast col-md-3 col-xs-12"
    >
      <div
        className="forecast-header"
      >
        <div
          className="day"
        >
          --
        </div>
        <div
          className="date"
        >
          --
        </div>
      </div>
      <div
        className="forecast-content"
      >
        <div
          className="forecast-icon"
        >
          <img
            src=""
            width={48}
          />
        </div>
        <div
          className="degree"
        >
          --
          <sup>
            o
          </sup>
          C
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12 col-md-offset-0 col-xs-4 col-xs-offset-2"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              Low:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
          <div
            className="col-md-12 col-xs-4"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              High:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      className="forecast col-md-3 col-xs-12"
    >
      <div
        className="forecast-header"
      >
        <div
          className="day"
        >
          --
        </div>
        <div
          className="date"
        >
          --
        </div>
      </div>
      <div
        className="forecast-content"
      >
        <div
          className="forecast-icon"
        >
          <img
            src=""
            width={48}
          />
        </div>
        <div
          className="degree"
        >
          --
          <sup>
            o
          </sup>
          C
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12 col-md-offset-0 col-xs-4 col-xs-offset-2"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              Low:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
          <div
            className="col-md-12 col-xs-4"
            style={
              Object {
                "padding": "0px",
              }
            }
          >
            <small>
              High:
               
              --
              <sup>
                o
              </sup>
              C
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
